### PR TITLE
[SINT-1547] Add NPM confusion analyzer

### PR DESCRIPTION
**Pull request by jamessteel123**
This adds an analyzer which detects NPM confusion events. Specifically it checks when the version's manifest in NPM does not match the content in the package.json in a manner that can change the behavior of the code, or mislead the consumer of the package (report bugs to the wrong place, link to incorrect documentation).

### CODE REVIEW

**Review by enelli — COMMENTED** [review]
Adding a short description to the [NPM metadata heuristics](tests/analyzer/metadata/test_npm_metadata_mismatch.py) would be a good addition.

File: tests/analyzer/metadata/resources/sample_project_info.py [hunk 1]
```diff
@@ -319,6 +319,7 @@
         }
       ]
     },
+
```
Comment (enelli) [inline tests/analyzer/metadata/resources/sample_project_info.py:322]: Remove line?

Comment (jamessteel123) [reply tests/analyzer/metadata/resources/sample_project_info.py]: Should be fixed now

**Review by enelli — APPROVED** [review]
LGTM

**Review by juliendoutre — APPROVED** [review]
Looks good to me! We tested this new check with @jamessteel123 on the https://www.npmjs.com/package/darcyclarke-manifest-pkg NPM package, identified as part of a malicious dependency campaign.

**Pull request merged**